File: Sesion 6/Marco teorico/17CÓMO SE SOBRECARGAN LOS OPERADORES UNARIOS/17.java
class Numero {
    private int valor;

    Numero(int valor) {
        this.valor = valor;
    }

    Numero negativo() {
        return new Numero(-this.valor);
    }

    int obtenerValor() {
        return this.valor;
    }
}

public class Main {
    public static void main(String[] args) {
        Numero num = new Numero(10);

        Numero negativo = num.negativo();

        System.out.println("Número original: " + num.obtenerValor());
        System.out.println("Número negativo: " + negativo.obtenerValor());
    }
}

File: Sesion 6/Marco teorico/17CÓMO SE SOBRECARGAN LOS OPERADORES UNARIOS/17.py
class Numero:
    def __init__(self, valor=0):
        self.valor = valor

    def __neg__(self):
        return Numero(-self.valor)

    def obtener_valor(self):
        return self.valor


num = Numero(10)

negativo = -num

print("Número original:", num.obtener_valor())
print("Número negativo:", negativo.obtener_valor())